daft/expressions/expressions.py

```diff
@@ -822,6 +822,35 @@ def extract(self, pattern: str | Expression, index: int = 0) -> Expression:
         pattern_expr = Expression._to_expression(pattern)
         return Expression._from_pyexpr(self._expr.utf8_extract(pattern_expr._expr, index))
 
+    def extract_all(self, pattern: str | Expression, index: int = 0) -> Expression:
+        r"""Extracts the specified match group from all regex matches in each string in a string column. If index is 0, the entire match is returned.
+        If the pattern does not match or the group does not exist, a null value is returned.
+
+        Example:
+            >>> df = daft.from_pydict({"x": ["123 456", "789 012", "345 678"]})
+            >>> df.with_column("matches", df["x"].str.extract_all(r"(\d+) (\d+)")).collect()
+            ╭─────────┬────────────╮
+            │ x       ┆ matches    │
+            │ ---     ┆ ---        │
+            │ Utf8    ┆ List[Utf8] │
+            ╞═════════╪════════════╡
+            │ 123 456 ┆ [123 456]  │
+            ├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
+            │ 789 012 ┆ [789 012]  │
+            ├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
+            │ 345 678 ┆ [345 678]  │
+            ╰─────────┴────────────╯
+
+        Args:
+            pattern: The regex pattern to extract
+            index: The index of the regex match group to extract
+
+        Returns:
+            Expression: a List[Utf8] expression with the extracted regex matches
```

Comment:
Can we also try adding a `See Also:` section here that links to `.str.extract`? Could be good to start doing it.